mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -1,78 +1,36 @@
 use std::{
     num::{NonZeroU64, NonZeroUsize},
-    path::{Path, PathBuf},
+    path::PathBuf,
     time::Duration,
     vec::Vec,
 };
 
-use config::{Config, ConfigError, Environment, File, FileFormat};
-use serde::Deserialize;
+use serde::{Deserialize, Deserializer};
 
 use mqtt_util::{CredentialProviderSettings, Credentials};
 
 use crate::persist::FlushOptions;
 
-pub const DEFAULTS: &str = include_str!("../config/default.json");
 const DEFAULT_UPSTREAM_PORT: &str = "8883";
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct BridgeSettings {
     upstream: Option<ConnectionSettings>,
-
     remotes: Vec<ConnectionSettings>,
-
-    messages: MessagesSettings,
-
     storage: StorageSettings,
 }
 
 impl BridgeSettings {
-    pub fn new() -> Result<Self, ConfigError> {
-        let mut config = Config::new();
-
-        config.merge(File::from_str(DEFAULTS, FileFormat::Json))?;
-        config.merge(Environment::new())?;
-
-        config.try_into()
-    }
-
-    pub fn from_file<P>(path: P) -> Result<Self, ConfigError>
-    where
-        P: AsRef<Path>,
-    {
-        let mut config = Config::new();
-
-        config.merge(File::from_str(DEFAULTS, FileFormat::Json))?;
-        config.merge(File::from(path.as_ref()))?;
-        config.merge(Environment::new())?;
-
-        config.try_into()
-    }
-
-    pub fn from_upstream_details(
```

Comment:
with a little refactoring and new ctor functions this has been moved to tests crate, since it is used only for tests.